src/main/java/software/amazon/awssdk/crt/auth/credentials/X509CredentialsProvider.java

```diff
@@ -126,7 +142,39 @@ private X509CredentialsProvider(X509CredentialsProviderBuilder builder) {
             throw new IllegalArgumentException("X509CredentialsProvider - clientBootstrap and tlsContext must be non null");
         }
 
-        long nativeHandle = x509CredentialsProviderNew(this, clientBootstrap.getNativeHandle(), tlsContext.getNativeHandle(), thingName.getBytes(UTF8), roleAlias.getBytes(UTF8), endpoint.getBytes(UTF8));
+        long proxyTlsContextHandle = 0;
+        String proxyHost = null;
+        int proxyPort = 0;
+        int proxyAuthorizationType = 0;
+        String proxyAuthorizationUsername = null;
+        String proxyAuthorizationPassword = null;
+        HttpProxyOptions proxyOptions = builder.getProxyOptions();
+        if (proxyOptions != null) {
+            TlsContext proxyTlsContext = proxyOptions.getTlsContext();
+            if (proxyTlsContext != null) {
+                proxyTlsContextHandle = proxyTlsContext.getNativeHandle();
+            }
+
+            proxyHost = proxyOptions.getHost();
+            proxyPort = proxyOptions.getPort();
+            proxyAuthorizationType = proxyOptions.getAuthorizationType().getValue();
+            proxyAuthorizationUsername = proxyOptions.getAuthorizationUsername();
+            proxyAuthorizationPassword = proxyOptions.getAuthorizationPassword();
+        }
+
+        long nativeHandle = x509CredentialsProviderNew(
+                this,
+                clientBootstrap.getNativeHandle(),
+                tlsContext.getNativeHandle(),
+                thingName.getBytes(UTF8),
+                roleAlias.getBytes(UTF8),
+                endpoint.getBytes(UTF8),
+                proxyHost,
```

Comment:
should this be serialized via getBytes() as well?

Reply:
We could; it would lead to a bit more refactor in the connection manager